flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/sink/TestIcebergSink.java

```diff
@@ -414,6 +422,49 @@ void testOperatorsUidNameWitUidSuffix() throws Exception {
     assertThat(secondTransformation.getName()).isEqualTo("data-ingestion");
   }
 
+  @TestTemplate
+  void testErrorOnNullForRequiredField() throws Exception {
+    Assumptions.assumeFalse(
+        format == FileFormat.ORC, "ORC file format supports null values even for required fields.");
+
+    Schema icebergSchema =
+        new Schema(
+            Types.NestedField.required(1, "id2", Types.IntegerType.get()),
+            Types.NestedField.required(2, "data2", Types.StringType.get()));
+    Table table2 =
+        CATALOG_EXTENSION
+            .catalog()
+            .createTable(
+                TableIdentifier.of(DATABASE, "t2"),
+                icebergSchema,
+                PartitionSpec.unpartitioned(),
+                ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, format.name()));
+
+    // Null out a required field
+    List<Row> rows = List.of(Row.of(42, null));
+
+    env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+    DataStream<Row> dataStream =
+        env.addSource(createBoundedSource(rows), ROW_TYPE_INFO).uid("mySourceId");
+
+    TableSchema flinkSchema = FlinkSchemaUtil.toSchema(icebergSchema);
+    IcebergSink.forRow(dataStream, flinkSchema)
+        .table(table2)
+        .tableLoader(tableLoader)
```

Comment:
technically this is incorrect, as this `TableLoader` loads the default table (not `table2`).

you can look for example in this class
```
TableLoader leftTableLoader =
        TableLoader.fromCatalog(CATALOG_EXTENSION.catalogLoader(), TableIdentifier.of("left"));
```

Reply:
Good catch, although not central to the test logic. Changed.